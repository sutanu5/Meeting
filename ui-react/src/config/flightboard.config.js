module.exports = {
  'board' : {
    'nextUp' : 'Next Up',
    'statusAvailable' : 'Open',
    'statusBusy' : 'Busy',
    'statusError' : 'Error'
  },

  'navbar' : {
    'title' : 'Conference Room Availability',
  },

  'roomFilter' : {
    'filterTitle' : 'Locations',
    'filterAllTitle' : 'All Conference Rooms',
  },
};
