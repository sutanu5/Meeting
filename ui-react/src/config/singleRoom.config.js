module.exports = {
  'nextUp' : 'Next Up',
  'statusAvailable' : 'Open',
  'statusBusy' : 'Busy',
  'upcomingTitle' : 'Upcoming',
};
