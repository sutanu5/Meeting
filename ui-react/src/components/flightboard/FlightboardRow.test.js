import FlightboardRow from './FlightboardRow';
import * as config from '../../config/flightboard.config.js';
import { MemoryRouter } from 'react-router-dom';

describe('Flightboard Row Component', () => {
  let room;

  beforeEach(() => {
    room = {
      'Roomlist':'Test Roomlist',
      'Name':'Test Room',
      'RoomAlias':'test-room',
      'Email':'email@email.com',
      'Appointments':[
         {
            'Subject':'Meeting Subject',
            'Organizer':'John Doe',
            'Start':1532966400000,
            'End':1533344400000
         }
      ],
      'Busy':true
    };
  });

  // all room.*

  it('renders correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });

  // roomStatusClass

  it('renders correct roomStatusClass when room.Busy is false', () => {
    room.Busy = false;
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const div = wrapper.find('.test-room .meeting-room .meeting-room-busy').exists();
    expect(div).not.toBeTruthy();
  });

  it('renders correct roomStatusClass when room.Busy is true', () => {
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const div = wrapper.find('.test-room .meeting-room .meeting-room-busy').exists();
    expect(div).toBeTruthy();
  });

  it('renders correct roomStatusClass when room.ErrorMessage is not empty', () => {
    room.ErrorMessage = 'Houston, we have a problem.';
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const div = wrapper.find('.test-room .meeting-room .meeting-room-error').exists();
    expect(div).toBeTruthy();
  });

  // /roomStatusClass
  // Status tests

  it('renders Status', () => {
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const status = wrapper.find('Status').exists();
    expect(status).toBeTruthy();
  });

  it('renders correct statusClass when room.Busy is false', () => {
    room.Busy = false;
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const div = wrapper.find('.meeting-open').exists();
    expect(div).toBeTruthy();
  });

  it('renders correct statusClass when room.Busy is true', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const div = wrapper.find('.meeting-busy').exists();
    expect(div).toBeTruthy();
  });

  it('renders correct statusClass when room.ErrorMessage is not empty', () => {
    room.ErrorMessage = 'Houston, we have a problem.';
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const div = wrapper.find('.meeting-error').exists();
    expect(div).toBeTruthy();
  });

  it('renders statusText correctly when room.Busy is true', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const status = wrapper.find('.test-room-meeting-status').text();
    expect(status).toBe(config.board.statusBusy);
  });

  it('renders statusText correctly when room.Busy is false', () => {
    room.Busy = false;
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const status = wrapper.find('.test-room-meeting-status').text();
    expect(status).toBe(config.board.statusAvailable);
  });

  it('renders statusText correctly when room.ErrorMessage is not empty', () => {
    room.ErrorMessage = 'Houston, we have a problem.';
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const status = wrapper.find('.test-room-meeting-status').text();
    expect(status).toBe(config.board.statusError);
  });

  it('displays busy when room.Busy is true', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const busy = wrapper.find('.meeting-busy').exists();
    expect(busy).toBeTruthy();
  });

  it('displays open when room.Busy is false', () => {
    room.Busy = false;

    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const open = wrapper.find('.meeting-open').exists();
    expect(open).toBeTruthy();
  });

  it('displays an error when room.ErrorMessage exists', () => {
    room.ErrorMessage = 'Houston, we have a problem.';

    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const error = wrapper.find('.meeting-error').exists();
    expect(error).toBeTruthy();
  });

  // /Status tests
  // Subject tests

  it('renders Subject', () => {
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const subject = wrapper.find('Subject').exists();
    expect(subject).toBeTruthy();
  });

  it('renders Subject empty if appointments do not exists', () => {
    room.Appointments = [];

    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const subject = wrapper.find('Subject').text();
    expect(subject).toEqual('');
  });

  it('renders correct meeting subject based on room.Appointments[0].Subject', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const subject = wrapper.find('.meeting-subject').text();
    expect(subject).toBe(room.Appointments[0].Subject);
  });

  // /Subject tests
  // Time tests

  it('renders Time', () => {
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const time = wrapper.find('Time').exists();
    expect(time).toBeTruthy();
  });

  it('renders Time empty if appointments do not exists', () => {
    room.Appointments = [];

    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const time = wrapper.find('Time').text();
    expect(time).toEqual('');
  });

  it('receives number from room.Appointments[0].Start', () => {
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const roomProps = wrapper.find('Time').props().room;
    expect(roomProps.Appointments[0].Start).toBeNumber(); 
  });

  it('receives number from room.Appointments[0].End', () => {
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const roomProps = wrapper.find('Time').props().room;
    expect(roomProps.Appointments[0].End).toBeNumber(); 
  });

  // /Time tests
  // Organizer tests

  it('renders Organizer', () => {
    const wrapper = shallow(<FlightboardRow room={room} filter='' />);

    const organizer = wrapper.find('Organizer').exists();
    expect(organizer).toBeTruthy();
  });

  it('renders Organizer empty if appointments do not exists', () => {
    room.Appointments = [];

    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const organizer = wrapper.find('Organizer').text();
    expect(organizer).toEqual('');
  });

  it('renders correct meeting organizer based on room.Appointments[0].Organizer', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const organizer = wrapper.find('.meeting-organizer').text();
    expect(organizer).toBe(room.Appointments[0].Organizer);
  });

  // /Organizer tests
  // FullScreenIcon tests

  it('renders FullScreenIcon Link', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const link = wrapper.find('.meeting-fullscreen a .fi-monitor').exists();
    expect(link).toBeTruthy();
  });

  it('does not render FullScreenIcon Link if room.ErrorMessage exists', () => {
    room.ErrorMessage = 'Houston, we have a problem.';

    const wrapper = mount(
      <MemoryRouter>
        <FlightboardRow room={room} filter='' />
      </MemoryRouter>
    );

    const link = wrapper.find('.meeting-fullscreen a .fi-monitor').exists();
    expect(link).toBeFalsy();
  });

  // /FullScreenIcon tests

  it('adds filter class based on roomlist', () => {
    const wrapper = shallow(<FlightboardRow room={room} filter='test-roomlist' />);

    const filterApplied = wrapper.find('.roomlist-test-roomlist').exists();
    expect(filterApplied).toBeTruthy();
  });
});
