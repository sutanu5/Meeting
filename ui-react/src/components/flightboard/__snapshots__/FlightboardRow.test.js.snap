// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flightboard Row Component renders correctly 1`] = `
<div
  className="meeting-room__row row-padder roomlist-test-roomlist"
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="medium-12 columns"
    >
      <div
        className="test-room meeting-room meeting-room-busy"
      >
        <div
          className="row valign-middle"
        >
          <div
            className="test-room-status meeting-room__status medium-2 columns"
          >
            <div
              className="test-room-meeting-status meeting-busy"
              title=""
            >
              Busy
            </div>
          </div>
          <div
            className="medium-3 columns"
          >
            <div
              className="test-room-name meeting-room__name"
            >
              Test Room
            </div>
          </div>
          <div
            className="medium-6 columns"
          >
            <div
              className="test-room-meeting-information"
            >
              <div>
                <span
                  className="test-room-meeting-upcoming meeting-upcoming"
                />
                <span
                  className="test-room-subject meeting-subject"
                >
                  Meeting Subject
                </span>
              </div>
            </div>
            <div
              className="test-room-time meeting-time"
            >
              9:00 AM - 6:00 PM
            </div>
            <div
              className="test-room-organizer meeting-organizer"
            >
              John Doe
            </div>
          </div>
          <div
            className="medium-1 columns"
          >
            <div
              className="meeting-fullscreen"
            >
              <a
                href="/single-room/test-room"
                onClick={[Function]}
                target="_blank"
              >
                <i
                  className="fi-monitor"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
